prime-router/src/main/kotlin/azure/SubmissionsFacade.kt

```diff
@@ -0,0 +1,118 @@
+package gov.cdc.prime.router.azure
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import gov.cdc.prime.router.ActionResponse
+import gov.cdc.prime.router.Metadata
+import gov.cdc.prime.router.Submission
+import gov.cdc.prime.router.SubmissionsProvider
+import java.time.OffsetDateTime
+
+/**
+ * Submissions / history API
+ * Contains all business logic regarding submissions and JSON serialization.
+ */
+
+/**
+ * TODO: Add test/SubmissionsFacadeTests
+ */
+class SubmissionsFacade(
+    private val metadata: Metadata,
+    private val db: DatabaseSubmissionsAccess = DatabaseSubmissionsAccess()
+) : SubmissionsProvider {
+    enum class AccessResult {
+        SUCCESS,
+        CREATED,
+        NOT_FOUND,
+        BAD_REQUEST
+    }
+
+    // Ignoring unknown props because we don't need them all. -DK
+    private val mapper = jacksonObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+    init {
+        // Format OffsetDateTime as an ISO string
+        mapper.registerModule(JavaTimeModule())
+        mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+    }
+
+    override fun findSubmissionsAsJson(
+        organizationName: String,
+        limit: String
+    ): String {
+        val result = findSubmissions(organizationName, limit)
+        return mapper.writeValueAsString(result)
+    }
+
+    private fun findSubmissions(organizationName: String, limit: String): List<SubmissionAPI> {
+        // TODO: VERIFY sendingOrg is being populated from the claim
+        val submissions = db.fetchSubmissions(organizationName, limit)
+
+        return submissions.map {
+            val actionResponse = mapper.readValue(it.actionResponse.toString(), ActionResponseAPI::class.java)
+            val result = SubmissionAPI(it.actionId, it.createdAt, it.sendingOrg, it.httpStatus, actionResponse)
+            result
+        }
+    }
+
+    companion object {
+        val metadata: Metadata by lazy {
```

Comment:
FYI: I made changes to how this is initialized, so either you or me need to merge this change in when it happens

Reply:
My change is now merged, so you will need to change this to
val metadata = Metadata.getInstance()